ansible_collections/arista/avd/roles/eos_designs/templates/network_services/logic.j2

```diff
@@ -85,6 +85,29 @@
                                                                            svi_profile.structured_config,
                                                                            svi_parent_profile.structured_config)}) %}
 {%                                 do tmp_svi.setdefault("bgp", {}).update({"structured_config": bgp_svi_struct_cfg}) %}
+{%                                 set tmp_svi_node = tmp_svi.pop("nodes", {}) | arista.avd.convert_dicts('name') |
+                                                                                 selectattr('name', 'arista.avd.defined', inventory_hostname) |
+                                                                                 first %}
+{%                                 if tmp_svi_node is arista.avd.defined %}
+{%                                     if tmp_svi.ip_address_virtual is arista.avd.defined and
+                                          tmp_svi_node.ip_address is arista.avd.defined and tmp_svi.ip_virtual_router_addresses is arista.avd.defined %}
+{%                                         do tmp_svi.update({"ip_address_virtual": none, "ip_address_virtual_secondaries": none}) %}
+{%                                     endif %}
+{%                                     if tmp_svi.ipv6_address_virtual is arista.avd.defined and
+                                          tmp_svi_node.ipv6_address is arista.avd.defined and tmp_svi.ipv6_virtual_router_addresses is arista.avd.defined %}
+{%                                         do tmp_svi.update({"ipv6_address_virtual": none, "ipv6_address_virtuals": none}) %}
+{%                                     endif %}
+{%                                     do tmp_svi_node.pop("structured_config", none) %}
+{%                                     do tmp_svi_node.pop("name") %}
+{%                                     do tmp_svi.update(tmp_svi_node) %}
+{%                                 else %}
+{%                                     if tmp_svi.ip_virtual_router_addresses is arista.avd.defined %}
+{%                                         do tmp_svi.update({"ip_virtual_router_addresses": none}) %}
```

Comment:
Instead of all these checks and modifications here, it would be simpler to understand, if you add the ip address condition to the places where we actually render something around VARP. (svis and static_routes)